packages/webapp/src/components/Form/NumberInput/index.tsx

```diff
@@ -56,13 +52,15 @@ export default function NumberInput({
   name,
   control,
   rules,
+  defaultValue,
+  className,
   onChange,
   onBlur,
   ...props
-}: NumberInputProps & RhfProps) {
-  const { field, fieldState } = useController({ name, control, rules, defaultValue: initialValue });
+}: NumberInputProps<T>) {
+  const { field, fieldState } = useController({ name, control, rules, defaultValue });
   const { inputProps, reset, numericValue, increment, decrement } = useNumberInput({
-    initialValue,
+    initialValue: defaultValue,
```

Comment:
The initial value type was giving an error so changed it use defaultValue prop that comes with react hook form